### ISSUE TITLE
`return-in-generator` (`B901`) - should not be reported on pytest hook wrappers

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#21881]
### Summary

```py
from collections.abc import Generator

from pytest import TestReport, hookimpl


@hookimpl(wrapper=True)
def pytest_runtest_makereport() -> Generator[None, TestReport, TestReport]:
    result = yield
    result.outcome = "passed"
    return result # error: return-in-generator (B901)
```

this is the intended usage of pytest hook wrappers. see https://docs.pytest.org/en/stable/how-to/writing_hook_functions.html#hook-wrappers-executing-around-other-hooks

this behavior was introduced in 0.14.8 (#21200)

### Version

0.14.8

**Comment by ntBre** [comment]
Ah thanks for the link to the docs! I guess that explains the ecosystem hits on pytest from the PR. It makes sense to me to special case functions decorated with `pytest.hookimpl`.

**Comment by assadyousuf** [comment]
Have a fix for this [here](https://github.com/astral-sh/ruff/pull/21931). Specifically looking for pytest hook wrappers before reporting `return-in-generator` @MichaReiser @ntBre
